src/mixer/playermanager.cpp

```diff
@@ -64,15 +64,15 @@ PlayerManager::PlayerManager(UserSettingsPointer pConfig,
                 ConfigKey("[Master]", "num_auxiliaries"), true, true)),
         m_pTrackAnalysisScheduler(TrackAnalysisScheduler::NullPointer()) {
     m_pCONumDecks->connectValueChangeRequest(this,
-            SLOT(slotChangeNumDecks(double)), Qt::DirectConnection);
+            [=](double value){slotChangeNumDecks(value);}, Qt::DirectConnection);
```

Comment:
We should enable here the normal member function pointer syntax. 
This is the Qt solution: 
```
    template <typename Func1, typename Func2>
    static inline QMetaObject::Connection connect(const typename QtPrivate::FunctionPointer<Func1>::Object *sender, Func1 signal,
                                     const typename QtPrivate::FunctionPointer<Func2>::Object *receiver, Func2 slot,
                                     Qt::ConnectionType type = Qt::AutoConnection)
```

I think it is something like 

```
template <typename Func>
bool ControlDoublePrivate::connectValueChangeRequest(
        const typename QtPrivate::FunctionPointer<Func>::Object *receiver, 
        Func slot, 
        Qt::ConnectionType type) {
```

Reply:
That is a better solution, thank you.

Reply:
I tried this but I have not gotten it to work:
```
In file included from src/track/keyutils.h:8,
                 from src/track/trackrecord.h:7,
                 from src/track/track.h:10,
                 from src/engine/cachingreader.h:16,
                 from src/engine/enginebuffer.h:25,
                 from src/engine/bpmcontrol.cpp:7:
src/control/controlproxy.h: In instantiation of ‘bool ControlProxy::connectValueChanged(const QObject*, Func, Qt::ConnectionType) [with Func = void (BpmControl::*)(double)]’:
src/control/controlproxy.h:97:58:   required from ‘bool ControlProxy::connectValueChanged(Func, Qt::ConnectionType) [with Func = void (BpmControl::*)(double)]’
src/engine/bpmcontrol.cpp:42:60:   required from here
src/control/controlproxy.h:79:13: error: no matching function for call to ‘ControlProxy::connect(ControlProxy*, void (ControlProxy::*)(double), const QObject*&, void (BpmControl::*&)(double), Qt::ConnectionType&)’
         if (!connect(this, &ControlProxy::valueChanged, receiver, method, scoConnection)) {
In file included from /usr/include/qt5/QtCore/QObject:1,
                 from src/control/controlobject.h:21,
                 from src/engine/bpmcontrol.cpp:3:
/usr/include/qt5/QtCore/qobject.h:228:43: note: candidate: ‘static QMetaObject::Connection QObject::connect(const typename QtPrivate::FunctionPointer<Func>::Object*, Func1, const typename QtPrivate::FunctionPointer<Func2>::Object*, Func2, Qt::ConnectionType) [with Func1 = void (ControlProxy::*)(double); Func2 = void (BpmControl::*)(double); typename QtPrivate::FunctionPointer<Func>::Object = ControlProxy; typename QtPrivate::FunctionPointer<Func2>::Object = BpmControl]’ <near match>
     static inline QMetaObject::Connection connect(const typename QtPrivate::FunctionPointer<Func1>::Object *sender, Func1 signal,
                                           ^~~~~~~
/usr/include/qt5/QtCore/qobject.h:228:43: note:   conversion of argument 3 would be ill-formed:
In file included from src/track/keyutils.h:8,
                 from src/track/trackrecord.h:7,
                 from src/track/track.h:10,
                 from src/engine/cachingreader.h:16,
                 from src/engine/enginebuffer.h:25,
                 from src/engine/bpmcontrol.cpp:7:
src/control/controlproxy.h:79:57: error: invalid conversion from ‘const QObject*’ to ‘const Object*’ {aka ‘const BpmControl*’} [-fpermissive]
         if (!connect(this, &ControlProxy::valueChanged, receiver, method, scoConnection)) {
                                                         ^~~~~~~~
In file included from /usr/include/qt5/QtCore/QObject:1,
                 from src/control/controlobject.h:21,
                 from src/engine/bpmcontrol.cpp:3:
/usr/include/qt5/QtCore/qobject.h:208:36: note: candidate: ‘static QMetaObject::Connection QObject::connect(const QObject*, const char*, const QObject*, const char*, Qt::ConnectionType)’
     static QMetaObject::Connection connect(const QObject *sender, const char *signal,
                                    ^~~~~~~
/usr/include/qt5/QtCore/qobject.h:208:36: note:   no known conversion for argument 2 from ‘void (ControlProxy::*)(double)’ to ‘const char*’
/usr/include/qt5/QtCore/qobject.h:211:36: note: candidate: ‘static QMetaObject::Connection QObject::connect(const QObject*, const QMetaMethod&, const QObject*, const QMetaMethod&, Qt::ConnectionType)’
     static QMetaObject::Connection connect(const QObject *sender, const QMetaMethod &signal,
                                    ^~~~~~~
/usr/include/qt5/QtCore/qobject.h:211:36: note:   no known conversion for argument 2 from ‘void (ControlProxy::*)(double)’ to ‘const QMetaMethod&’
/usr/include/qt5/QtCore/qobject.h:463:32: note: candidate: ‘QMetaObject::Connection QObject::connect(const QObject*, const char*, const char*, Qt::ConnectionType) const’
 inline QMetaObject::Connection QObject::connect(const QObject *asender, const char *asignal,
                                ^~~~~~~
/usr/include/qt5/QtCore/qobject.h:463:32: note:   candidate expects 4 arguments, 5 provided
/usr/include/qt5/QtCore/qobject.h:260:13: note: candidate: ‘template<class Func1, class Func2> static typename std::enable_if<((int)(QtPrivate::FunctionPointer<Func2>::ArgumentCount) >= 0), QMetaObject::Connection>::type QObject::connect(const typename QtPrivate::FunctionPointer<Func>::Object*, Func1, Func2)’
             connect(const typename QtPrivate::FunctionPointer<Func1>::Object *sender, Func1 signal, Func2 slot)
             ^~~~~~~
/usr/include/qt5/QtCore/qobject.h:260:13: note:   template argument deduction/substitution failed:
In file included from src/track/keyutils.h:8,
                 from src/track/trackrecord.h:7,
                 from src/track/track.h:10,
                 from src/engine/cachingreader.h:16,
                 from src/engine/enginebuffer.h:25,
                 from src/engine/bpmcontrol.cpp:7:
src/control/controlproxy.h:79:13: note:   candidate expects 3 arguments, 5 provided
         if (!connect(this, &ControlProxy::valueChanged, receiver, method, scoConnection)) {
In file included from /usr/include/qt5/QtCore/QObject:1,
                 from src/control/controlobject.h:21,
                 from src/engine/bpmcontrol.cpp:3:
/usr/include/qt5/QtCore/qobject.h:269:13: note: candidate: ‘template<class Func1, class Func2> static typename std::enable_if<(((int)(QtPrivate::FunctionPointer<Func2>::ArgumentCount) >= 0) && (! QtPrivate::FunctionPointer<Func2>::IsPointerToMemberFunction)), QMetaObject::Connection>::type QObject::connect(const typename QtPrivate::FunctionPointer<Func>::Object*, Func1, const QObject*, Func2, Qt::ConnectionType)’
             connect(const typename QtPrivate::FunctionPointer<Func1>::Object *sender, Func1 signal, const QObject *context, Func2 slot,
             ^~~~~~~
/usr/include/qt5/QtCore/qobject.h:269:13: note:   template argument deduction/substitution failed:
/usr/include/qt5/QtCore/qobject.h: In substitution of ‘template<class Func1, class Func2> static typename std::enable_if<(((int)(QtPrivate::FunctionPointer<Func2>::ArgumentCount) >= 0) && (! QtPrivate::FunctionPointer<Func2>::IsPointerToMemberFunction)), QMetaObject::Connection>::type QObject::connect(const typename QtPrivate::FunctionPointer<Func>::Object*, Func1, const QObject*, Func2, Qt::ConnectionType) [with Func1 = void (ControlProxy::*)(double); Func2 = void (BpmControl::*)(double)]’:
src/control/controlproxy.h:79:13:   required from ‘bool ControlProxy::connectValueChanged(const QObject*, Func, Qt::ConnectionType) [with Func = void (BpmControl::*)(double)]’
src/control/controlproxy.h:97:58:   required from ‘bool ControlProxy::connectValueChanged(Func, Qt::ConnectionType) [with Func = void (BpmControl::*)(double)]’
src/engine/bpmcontrol.cpp:42:60:   required from here
/usr/include/qt5/QtCore/qobject.h:269:13: error: no type named ‘type’ in ‘struct std::enable_if<false, QMetaObject::Connection>’
src/control/controlproxy.h: In instantiation of ‘bool ControlProxy::connectValueChanged(const QObject*, Func, Qt::ConnectionType) [with Func = void (BpmControl::*)(double)]’:
src/control/controlproxy.h:97:58:   required from ‘bool ControlProxy::connectValueChanged(Func, Qt::ConnectionType) [with Func = void (BpmControl::*)(double)]’
src/engine/bpmcontrol.cpp:42:60:   required from here
/usr/include/qt5/QtCore/qobject.h:300:13: note: candidate: ‘template<class Func1, class Func2> static typename std::enable_if<(QtPrivate::FunctionPointer<Func2>::ArgumentCount == -1), QMetaObject::Connection>::type QObject::connect(const typename QtPrivate::FunctionPointer<Func>::Object*, Func1, Func2)’
             connect(const typename QtPrivate::FunctionPointer<Func1>::Object *sender, Func1 signal, Func2 slot)
             ^~~~~~~
/usr/include/qt5/QtCore/qobject.h:300:13: note:   template argument deduction/substitution failed:
In file included from src/track/keyutils.h:8,
                 from src/track/trackrecord.h:7,
                 from src/track/track.h:10,
                 from src/engine/cachingreader.h:16,
                 from src/engine/enginebuffer.h:25,
                 from src/engine/bpmcontrol.cpp:7:
src/control/controlproxy.h:79:13: note:   candidate expects 3 arguments, 5 provided
         if (!connect(this, &ControlProxy::valueChanged, receiver, method, scoConnection)) {
In file included from /usr/include/qt5/QtCore/QObject:1,
                 from src/control/controlobject.h:21,
                 from src/engine/bpmcontrol.cpp:3:
/usr/include/qt5/QtCore/qobject.h:308:13: note: candidate: ‘template<class Func1, class Func2> static typename std::enable_if<(QtPrivate::FunctionPointer<Func2>::ArgumentCount == -1), QMetaObject::Connection>::type QObject::connect(const typename QtPrivate::FunctionPointer<Func>::Object*, Func1, const QObject*, Func2, Qt::ConnectionType)’
             connect(const typename QtPrivate::FunctionPointer<Func1>::Object *sender, Func1 signal, const QObject *context, Func2 slot,
             ^~~~~~~
/usr/include/qt5/QtCore/qobject.h:308:13: note:   template argument deduction/substitution failed:
/usr/include/qt5/QtCore/qobject.h: In substitution of ‘template<class Func1, class Func2> static typename std::enable_if<(QtPrivate::FunctionPointer<Func2>::ArgumentCount == -1), QMetaObject::Connection>::type QObject::connect(const typename QtPrivate::FunctionPointer<Func>::Object*, Func1, const QObject*, Func2, Qt::ConnectionType) [with Func1 = void (ControlProxy::*)(double); Func2 = void (BpmControl::*)(double)]’:
src/control/controlproxy.h:79:13:   required from ‘bool ControlProxy::connectValueChanged(const QObject*, Func, Qt::ConnectionType) [with Func = void (BpmControl::*)(double)]’
src/control/controlproxy.h:97:58:   required from ‘bool ControlProxy::connectValueChanged(Func, Qt::ConnectionType) [with Func = void (BpmControl::*)(double)]’
src/engine/bpmcontrol.cpp:42:60:   required from here
/usr/include/qt5/QtCore/qobject.h:308:13: error: no type named ‘type’ in ‘struct std::enable_if<false, QMetaObject::Connection>’
```